app/controllers/answers_controller.rb

```diff
@@ -0,0 +1,54 @@
+class AnswersController < ApplicationController
+
+  def create
+    @question = Question.find(params[:question_id])
+    @answer = @question.answers.build(answer_params)
+    if @answer.save
+      flash[:notice] = "Your answer was posted!"
+      redirect_to @question
+    else
+      flash.now[:notice] = "Your answer was not posted correct."
+      # @question = Question.new
+      render "questions/show"
+    end
+  end
+
+  def edit
+    @answer = Answer.find(params[:id])
+    @question = @answer.question
+  end
+
+  def update
+      @answer =  Answer.find(params[:id])
```

Comment:
Don't hijack update to do the best answer action. Add in routes.rb something like 
```
member do
  post 'best_answer'
end
```
and rails will generate a route to an action called best_answer